toolchain/check/implicit_type_impls.h

```diff
@@ -0,0 +1,18 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_IMPLICIT_TYPE_IMPLS_H_
+#define CARBON_TOOLCHAIN_CHECK_IMPLICIT_TYPE_IMPLS_H_
+
+#include "toolchain/check/context.h"
+
+namespace Carbon::Check {
+
+// Constructs `impl <class> as Destroy`, with appropriate implementation based
```

Comment:
```suggestion
// Constructs `impl <class> as Destroy { ... }`, with appropriate implementation based
```

Just to be a bit clearer that this doesn't just declare the impl, it defines it.